README.md

```diff
@@ -99,40 +104,65 @@ protect-ffi/
-| `lib.rs`       | Entry point for the Rust source code.                                                                                                          |
-| `platforms/`   | The directory containing distributions of the binary addon backend for each platform supported by this library.                          |
-| `package.json` | The npm [manifest file](https://docs.npmjs.com/cli/v7/configuring-npm/package-json), which informs the `npm` command.                    |
-| `target/`      | Binary artifacts generated by the Rust build.                                                                                            |
+| Entry                | Purpose                                                                                                                            |
+| -------------------- | ---------------------------------------------------------------------------------------------------------------------------------- |
+| `Cargo.toml`         | The Cargo [manifest file](https://doc.rust-lang.org/cargo/reference/manifest.html), which informs the `cargo` command.             |
+| `README.md`          | This file.                                                                                                                         |
+| `integration-tests/` | The directory containing integration tests.                                              |
+| `lib/`               | The directory containing the generated output from [tsc](https://typescriptlang.org).                                              |
+| `src/`               | The directory containing the TypeScript source files.                                                                              |
+| `index.mts`          | Entry point for when this library is loaded via [ESM `import`](https://nodejs.org/api/esm.html#modules-ecmascript-modules) syntax. |
+| `index.cts`          | Entry point for when this library is loaded via [CJS `require`](https://nodejs.org/api/modules.html#requireid).                    |
+| `crates/`            | The directory tree containing the Rust source code for the project.                                                                |
+| `lib.rs`             | Entry point for the Rust source code.                                                                                              |
+| `platforms/`         | The directory containing distributions of the binary addon backend for each platform supported by this library.                    |
+| `package.json`       | The npm [manifest file](https://docs.npmjs.com/cli/v7/configuring-npm/package-json), which informs the `npm` command.              |
+| `target/`            | Binary artifacts generated by the Rust build.                                                                                      |
+
+## Integration tests
+
+Integration tests live in the `./integration-tests` directory.
```

Comment:
Would it make more sense to put these tests in `tests/integration/javascript/`?

That way it more closely follows the [Rust conventions for integration tests](https://doc.rust-lang.org/book/ch11-03-test-organization.html#integration-tests), and makes it explicit that these are JavaScript tests?

Only downside of this I can see is `cargo test` may not like there not being Rust tests in the `tests/` directory.

Reply:
I think I'd find `tests/integration/javascript` slightly more surprising because I don't think I'd put other tests under the `./tests` dir, but the nested structure implies that we would. I think I'd but put Rust tests directly in `crates/protect-ffi` (and there are already some tests there) and I'd probably put unit tests for JS directly in `./src`.